extensions/mssql/src/objectExplorerNodeProvider/fileSources.ts

```diff
@@ -181,7 +181,7 @@ export class HdfsFileSource implements IFileSource {
 	public readFile(path: string, maxBytes?: number): Promise<Buffer> {
 		return new Promise((resolve, reject) => {
 			let error: HdfsError = undefined;
-			let remoteFileStream = this.client.createReadStream(path);
+			let remoteFileStream = this.client.createReadStream(path, maxBytes ? {'length': maxBytes} : {});
```

Comment:
So, I used the meter code below this to handle going over the max bytes. Did I miss that we could just specify a "stop after N bytes" option in the request args instead? How did you learn about this?